useScroll/Scrollbar.js

```diff
@@ -1,112 +1,131 @@
 
+const useThemeScrollbar = (props) => {
+	const {
+		restProps,
+		scrollbarProps,
+		scrollbarTrackProps
+	} = useScrollbarBase(props);
+
+	const {
+		cbAlertScrollbarTrack,
+		onNextScroll,
+		onPrevScroll,
+		rtl,
+		...rest
+	} = restProps;
+
+	const {vertical} = scrollbarTrackProps;
+
+	return {
+		restProps: rest,
+		scrollbarProps,
+		scrollbarButtonsProps: {
+			onNextScroll,
+			onPrevScroll,
+			rtl,
+			vertical
+		},
+		scrollbarTrackProps: {
+			...scrollbarTrackProps,
+			cbAlertScrollbarTrack
+		}
+	};
+};
+
 /**
- * An Agate-styled scroller base component.
+ * An Agate-styled scrollbar base component.
  *
  * @class ScrollbarBase
  * @memberof agate/useScroll
  * @extends ui/ScrollbarBase
  * @ui
  * @private
  */
-const ScrollbarBase = memo(forwardRef((props, ref) => {
-	// Refs
-	const scrollbarRef = useRef();
+const ScrollbarBase = memo((props) => {
 	const scrollButtonsRef = useRef();
-	// render
-	const {cbAlertThumb, clientSize, corner, vertical, ...rest} = props;
 
-	useImperativeHandle(ref, () => {
-		const {getContainerRef, showThumb, startHidingThumb, update: uiUpdate} = scrollbarRef.current;
+	const {
+		restProps,
+		scrollbarProps,
+		scrollbarButtonsProps,
+		scrollbarTrackProps
+	} = useThemeScrollbar(props);
+
+	useEffect(() => {
+		const {scrollbarHandle} = props;
+		const {update: uiUpdate} = scrollbarHandle.current;
 		const {focusOnButton, isOneOfScrollButtonsFocused, updateButtons} = scrollButtonsRef.current;
 
-		return {
-			focusOnButton,
-			get uiScrollbarContainer () {
-				return getContainerRef();
-			},
-			isOneOfScrollButtonsFocused,
-			showThumb,
-			startHidingThumb,
-			uiUpdate,
-			update: (bounds) => {
-				updateButtons(bounds);
-				uiUpdate(bounds);
-			}
+		scrollbarHandle.current.update = (bounds) => {
+			updateButtons(bounds);
+			uiUpdate(bounds);
 		};
-	}, [scrollbarRef, scrollButtonsRef]);
+
+		scrollbarHandle.current.focusOnButton = focusOnButton;
+		scrollbarHandle.current.isOneOfScrollButtonsFocused = isOneOfScrollButtonsFocused;
+	});
 
 	return (
-		<UiScrollbarBase
-			clientSize={clientSize}
-			corner={corner}
-			css={componentCss}
-			ref={scrollbarRef}
-			vertical={vertical}
-			childRenderer={({thumbRef}) => { // eslint-disable-line react/jsx-no-bind
-				return (
-					<ScrollButtons
-						{...rest}
-						ref={scrollButtonsRef}
-						vertical={vertical}
-						thumbRenderer={() => { // eslint-disable-line react/jsx-no-bind
-							return (
-								<ScrollThumb
-									cbAlertThumb={cbAlertThumb}
-									key="thumb"
-									ref={thumbRef}
-									vertical={vertical}
-								/>
-							);
-						}}
-					/>
-				);
-			}}
-		/>
+		<div {...scrollbarProps}>
+			<ScrollButtons
+				{...restProps}
+				{...scrollbarButtonsProps}
+				ref={scrollButtonsRef}
+				scrollbarTrackRenderer={() => { // eslint-disable-line react/jsx-no-bind
+					return (
+						<ScrollbarTrack
+							{...scrollbarTrackProps}
+							key="scrollbarTrack"
```

Comment:
I might be better to add the comment why we need the key property here as you were curious before.

Reply:
I think it's okay